lib/vanagon/component.rb

```diff
@@ -140,10 +141,13 @@ def get_build_dir
     #
     # @param workdir [String] working directory to put the source into
     def get_sources(workdir)
-      @sources.each do |source|
-        cur_source = Vanagon::Component::Source.source(source.url, { :ref => source.ref, :sum => source.sum }, workdir)
-        cur_source.fetch
-        cur_source.verify
+      sources.each do |source|
```

Comment:
`attr_accessor` defines both the getter and setter methods, and directly translates those methods to the instance variable if no overriding method is defined. So calling `sources` here is actually calling the method, not the instance variable (`@sources`). You would only call the instance variable directly if you were actually doing something that would change the state of the value (beyond calling it or assigning it).